### DISCUSSION TITLE
Searching for Files in Same Directory as Julia File

### DISCUSSION THREAD

**Discussion by zapif** [OxygenFramework/Oxygen.jl#34] [Q&A] [answered]
I am really liking this package so far but I have ran into an issue when dealing with serving static files or returning them in the response. It looks like instead of searching for the files in the same directory as the Julia file, it instead looks in another directory which results in the file not being found.

In this code snippet the Julia file is being run in the directory /home/user/Documents/svelte/api/backend, and instead it looks for the file in the directory that I was testing the demo in.

`curl localhost:8081/api/zoo/files/test.txt`

```Starting server: http://127.0.0.1:8081
┌ Error: ERROR: 
│   exception =
│    SystemError: opening file "/home/user/Documents/julia/Oxygen/test.txt": No such file or directory
│    Stacktrace:
│      [1] systemerror(p::String, errno::Int32; extrainfo::Nothing)
│        @ Base ./error.jl:174
│      [2] #systemerror#68
│        @ ./error.jl:173 [inlined]
│      [3] systemerror
│        @ ./error.jl:173 [inlined]
│      [4] open(fname::String; lock::Bool, read::Nothing, write::Nothing, create::Nothing, truncate::Nothing, append::Nothing)
│        @ Base ./iostream.jl:293
│      [5] open
│        @ ./iostream.jl:282 [inlined]
│      [6] file(filename::String)
│        @ Oxygen.FileUtil ~/.julia/packages/Oxygen/E9kOd/src/fileutil.jl:14
│      [7] (::var"#17#18")(req::HTTP.Messages.Request, name::String)
│        @ Main ~/Documents/svelte/api/backend/oxygen.jl:49
│      [8] (::Oxygen.var"#145#157")(req::HTTP.Messages.Request)
│        @ Oxygen ~/.julia/packages/Oxygen/E9kOd/src/Oxygen.jl:414
│      [9] handle(::HTTP.Handlers.RequestHandlerFunction{Oxygen.var"#145#157"}, ::HTTP.Messages.Request)
│        @ HTTP.Handlers ~/.julia/packages/HTTP/aTjcj/src/Handlers.jl:254
│     [10] handle(::HTTP.Handlers.Router{Symbol("##291")}, ::HTTP.Messages.Request)
│        @ HTTP.Handlers ~/.julia/packages/HTTP/aTjcj/src/Handlers.jl:473
│     [11] DefaultHandler(req::HTTP.Messages.Request)
│        @ Oxygen ~/.julia/packages/Oxygen/E9kOd/src/Oxygen.jl:98
│     [12] CorsHandler(req::HTTP.Messages.Request, defaultHandler::typeof(Oxygen.DefaultHandler))
│        @ Main ~/Documents/svelte/api/backend/oxygen.jl:65
│     [13] #19
│        @ ~/Documents/svelte/api/backend/oxygen.jl:74 [inlined]
│     [14] #5
│        @ ~/.julia/packages/Oxygen/E9kOd/src/Oxygen.jl:42 [inlined]
│     [15] handle
│        @ ~/.julia/packages/HTTP/aTjcj/src/Handlers.jl:254 [inlined]
│     [16] handle(::HTTP.Handlers.RequestHandlerFunction{Oxygen.var"#5#6"{var"#19#20"}}, ::HTTP.Streams.Stream{HTTP.Messages.Request, HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}})
│        @ HTTP.Handlers ~/.julia/packages/HTTP/aTjcj/src/Handlers.jl:277
│     [17] #4
│        @ ~/.julia/packages/HTTP/aTjcj/src/Handlers.jl:351 [inlined]
│     [18] macro expansion
│        @ ~/.julia/packages/HTTP/aTjcj/src/Servers.jl:415 [inlined]
│     [19] (::HTTP.Servers.var"#13#14"{HTTP.Handlers.var"#4#5"{HTTP.Handlers.RequestHandlerFunction{Oxygen.var"#5#6"{var"#19#20"}}}, HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}, HTTP.Servers.Server{Nothing, Sockets.TCPServer}, HTTP.Streams.Stream{HTTP.Messages.Request, HTTP.ConnectionPool.Transaction{Sockets.TCPSocket}}})()
│        @ HTTP.Servers ./task.jl:429
└ @ Oxygen ~/.julia/packages/Oxygen/E9kOd/src/Oxygen.jl:419```

**Comment by ndortega** [comment, marked as answer]
Hey zapif, thanks for using this package!

Did you launch the julia repl from /home/user/Documents/svelte/api/backend ? At the moment, you can only host static files from inside the same project directory. If you were using the same repl session as the one to test the demo, then that file would be way outside the current project directory

**Reply by zapif** [reply]
I am running the Julia file from the backend directory by doing julia oxygen.jl and double checked to make sure there are no other repls running at the moment, and I am still getting the issue with it looking in that backend directory as well for the text file.

I did some debugging and found that that in fileutil.jl, the current_directory variable is listing the incorrect directory inside the file function. I moved the assignment of current_directory inside the function and now it seems to be getting the correct directory.

```

    """
        file(folder::String, filename::String)

    Read in a file as a String (from the projects current working directory)
    """
    function file(filename::String)
        current_directory = pwd()
        path = joinpath(current_directory, filename)
        return read(open(path), String)
    end
```

**Reply by ndortega** [reply]
I finally see what you mean! I'm also running into the same problem. I'm thinking of removing the `current_directory = pwd()` at the top and simplifying the file() function like this. I'd also recommend dropping the other redundant file() function below this one. 

This is a critical bug, we'll need to get this fixed asap. Feel free to open a pr with the changes yourself (if you want) or leave it to me to fix later today.

Thanks for testing this out and opening this discussion!

```julia 
    """
        file(filepath::String)

    Read in a file as a String
    """
    function file(filepath::String)
        return read(open(filepath), String)
    end

```

**Reply by zapif** [reply]
Great! I submitted the pr with these changes.
